commit 4668c7a2940d134bea50058e138591b97485c5da
Author: Akinobu Mita <akinobu.mita@gmail.com>
Date:   Mon Mar 27 23:37:32 2023 +0900

    fault-inject: allow configuration via configfs
    
    This provides a helper function to allow configuration of fault-injection
    for configfs-based drivers.
    
    The config items created by this function have the same interface as the
    one created under debugfs by fault_create_debugfs_attr().
    
    Signed-off-by: Akinobu Mita <akinobu.mita@gmail.com>
    Link: https://lore.kernel.org/r/20230327143733.14599-2-akinobu.mita@gmail.com
    Signed-off-by: Jens Axboe <axboe@kernel.dk>

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index c8b379e2e9ad..e700b29d7756 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -1958,9 +1958,20 @@ config FAIL_SUNRPC
 	  Provide fault-injection capability for SunRPC and
 	  its consumers.
 
+config FAULT_INJECTION_CONFIGFS
+	bool "Configfs interface for fault-injection capabilities"
+	depends on FAULT_INJECTION && CONFIGFS_FS
+	help
+	  This option allows configfs-based drivers to dynamically configure
+	  fault-injection via configfs.  Each parameter for driver-specific
+	  fault-injection can be made visible as a configfs attribute in a
+	  configfs group.
+
+
 config FAULT_INJECTION_STACKTRACE_FILTER
 	bool "stacktrace filter for fault-injection capabilities"
-	depends on FAULT_INJECTION_DEBUG_FS && STACKTRACE_SUPPORT
+	depends on FAULT_INJECTION
+	depends on (FAULT_INJECTION_DEBUG_FS || FAULT_INJECTION_CONFIGFS) && STACKTRACE_SUPPORT
 	select STACKTRACE
 	depends on FRAME_POINTER || MIPS || PPC || S390 || MICROBLAZE || ARM || ARC || X86
 	help